File: services/src/main/java/com/liferay/mobile/sdk/v62/mbcategory/MBCategoryService.java
package com.liferay.mobile.sdk.v62.mbcategory;

import com.liferay.mobile.sdk.Call;
import com.liferay.mobile.sdk.annotation.Param;
import com.liferay.mobile.sdk.annotation.Path;
import com.liferay.mobile.sdk.http.Response;
import java.lang.Integer;
import java.lang.String;
import org.json.JSONArray;
import org.json.JSONObject;

@Path("/mbcategory")
public interface MBCategoryService {
  @Path("/add-category")
  Call<JSONObject> addCategory(@Param(name = "userId") long userId, @Param(name = "parentCategoryId") long parentCategoryId, @Param(name = "name") String name, @Param(name = "description") String description, @Param(name = "serviceContext", className = "com.liferay.portal.service.ServiceContext") JSONObject serviceContext);

  @Path("/add-category")
  Call<JSONObject> addCategory(@Param(name = "parentCategoryId") long parentCategoryId, @Param(name = "name") String name, @Param(name = "description") String description, @Param(name = "displayStyle") String displayStyle, @Param(name = "emailAddress") String emailAddress, @Param(name = "inProtocol") String inProtocol, @Param(name = "inServerName") String inServerName, @Param(name = "inServerPort") int inServerPort, @Param(name = "inUseSSL") boolean inUseSSL, @Param(name = "inUserName") String inUserName, @Param(name = "inPassword") String inPassword, @Param(name = "inReadInterval") int inReadInterval, @Param(name = "outEmailAddress") String outEmailAddress, @Param(name = "outCustom") boolean outCustom, @Param(name = "outServerName") String outServerName, @Param(name = "outServerPort") int outServerPort, @Param(name = "outUseSSL") boolean outUseSSL, @Param(name = "outUserName") String outUserName, @Param(name = "outPassword") String outPassword, @Param(name = "mailingListActive") boolean mailingListActive, @Param(name = "allowAnonymousEmail") boolean allowAnonymousEmail, @Param(name = "serviceContext", className = "com.liferay.portal.service.ServiceContext") JSONObject serviceContext);

  @Path("/delete-category")
  Call<Response> deleteCategory(@Param(name = "categoryId") long categoryId, @Param(name = "includeTrashedEntries") boolean includeTrashedEntries);

  @Path("/delete-category")
  Call<Response> deleteCategory(@Param(name = "groupId") long groupId, @Param(name = "categoryId") long categoryId);

  @Path("/get-categories")
  Call<JSONArray> getCategories(@Param(name = "groupId") long groupId);

  @Path("/get-categories")
  Call<JSONArray> getCategories(@Param(name = "groupId") long groupId, @Param(name = "status") int status);

  @Path("/get-categories")
  Call<JSONArray> getCategories(@Param(name = "groupId") long groupId, @Param(name = "parentCategoryId") long parentCategoryId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-categories")
  Call<JSONArray> getCategories(@Param(name = "groupId") long groupId, @Param(name = "parentCategoryIds") JSONArray parentCategoryIds, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-categories")
  Call<JSONArray> getCategories(@Param(name = "groupId") long groupId, @Param(name = "parentCategoryId") long parentCategoryId, @Param(name = "status") int status, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-categories")
  Call<JSONArray> getCategories(@Param(name = "groupId") long groupId, @Param(name = "parentCategoryIds") JSONArray parentCategoryIds, @Param(name = "status") int status, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-categories-count")
  Call<Integer> getCategoriesCount(@Param(name = "groupId") long groupId, @Param(name = "parentCategoryId") long parentCategoryId);

  @Path("/get-categories-count")
  Call<Integer> getCategoriesCount(@Param(name = "groupId") long groupId, @Param(name = "parentCategoryIds") JSONArray parentCategoryIds);

  @Path("/get-categories-count")
  Call<Integer> getCategoriesCount(@Param(name = "groupId") long groupId, @Param(name = "parentCategoryId") long parentCategoryId, @Param(name = "status") int status);

  @Path("/get-categories-count")
  Call<Integer> getCategoriesCount(@Param(name = "groupId") long groupId, @Param(name = "parentCategoryIds") JSONArray parentCategoryIds, @Param(name = "status") int status);

  @Path("/get-category")
  Call<JSONObject> getCategory(@Param(name = "categoryId") long categoryId);

  @Path("/get-category-ids")
  Call<JSONArray> getCategoryIds(@Param(name = "groupId") long groupId, @Param(name = "categoryId") long categoryId);

  @Path("/get-subcategory-ids")
  Call<JSONArray> getSubcategoryIds(@Param(name = "categoryIds") JSONArray categoryIds, @Param(name = "groupId") long groupId, @Param(name = "categoryId") long categoryId);

  @Path("/get-subscribed-categories")
  Call<JSONArray> getSubscribedCategories(@Param(name = "groupId") long groupId, @Param(name = "userId") long userId, @Param(name = "start") int start, @Param(name = "end") int end);

  @Path("/get-subscribed-categories-count")
  Call<Integer> getSubscribedCategoriesCount(@Param(name = "groupId") long groupId, @Param(name = "userId") long userId);

  @Path("/move-category")
  Call<JSONObject> moveCategory(@Param(name = "categoryId") long categoryId, @Param(name = "parentCategoryId") long parentCategoryId, @Param(name = "mergeWithParentCategory") boolean mergeWithParentCategory);

  @Path("/move-category-from-trash")
  Call<JSONObject> moveCategoryFromTrash(@Param(name = "categoryId") long categoryId, @Param(name = "newCategoryId") long newCategoryId);

  @Path("/move-category-to-trash")
  Call<JSONObject> moveCategoryToTrash(@Param(name = "categoryId") long categoryId);

  @Path("/restore-category-from-trash")
  Call<Response> restoreCategoryFromTrash(@Param(name = "categoryId") long categoryId);

  @Path("/subscribe-category")
  Call<Response> subscribeCategory(@Param(name = "groupId") long groupId, @Param(name = "categoryId") long categoryId);

  @Path("/unsubscribe-category")
  Call<Response> unsubscribeCategory(@Param(name = "groupId") long groupId, @Param(name = "categoryId") long categoryId);

  @Path("/update-category")
  Call<JSONObject> updateCategory(@Param(name = "categoryId") long categoryId, @Param(name = "parentCategoryId") long parentCategoryId, @Param(name = "name") String name, @Param(name = "description") String description, @Param(name = "displayStyle") String displayStyle, @Param(name = "emailAddress") String emailAddress, @Param(name = "inProtocol") String inProtocol, @Param(name = "inServerName") String inServerName, @Param(name = "inServerPort") int inServerPort, @Param(name = "inUseSSL") boolean inUseSSL, @Param(name = "inUserName") String inUserName, @Param(name = "inPassword") String inPassword, @Param(name = "inReadInterval") int inReadInterval, @Param(name = "outEmailAddress") String outEmailAddress, @Param(name = "outCustom") boolean outCustom, @Param(name = "outServerName") String outServerName, @Param(name = "outServerPort") int outServerPort, @Param(name = "outUseSSL") boolean outUseSSL, @Param(name = "outUserName") String outUserName, @Param(name = "outPassword") String outPassword, @Param(name = "mailingListActive") boolean mailingListActive, @Param(name = "allowAnonymousEmail") boolean allowAnonymousEmail, @Param(name = "mergeWithParentCategory") boolean mergeWithParentCategory, @Param(name = "serviceContext", className = "com.liferay.portal.service.ServiceContext") JSONObject serviceContext);
}

File: windows/Liferay.SDK/Service/V62/MBCategory/MBCategoryService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V62.MBCategory
{
	public class MBCategoryService : ServiceBase
	{
		public MBCategoryService(ISession session)
			: base(session)
		{
		}

		public async Task<IDictionary<string, object>> AddCategoryAsync(long parentCategoryId, string name, string description, string displayStyle, string emailAddress, string inProtocol, string inServerName, int inServerPort, bool inUseSSL, string inUserName, string inPassword, int inReadInterval, string outEmailAddress, bool outCustom, string outServerName, int outServerPort, bool outUseSSL, string outUserName, string outPassword, bool mailingListActive, bool allowAnonymousEmail, IDictionary<string, object> serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("parentCategoryId", parentCategoryId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("displayStyle", displayStyle);
			_parameters.Add("emailAddress", emailAddress);
			_parameters.Add("inProtocol", inProtocol);
			_parameters.Add("inServerName", inServerName);
			_parameters.Add("inServerPort", inServerPort);
			_parameters.Add("inUseSSL", inUseSSL);
			_parameters.Add("inUserName", inUserName);
			_parameters.Add("inPassword", inPassword);
			_parameters.Add("inReadInterval", inReadInterval);
			_parameters.Add("outEmailAddress", outEmailAddress);
			_parameters.Add("outCustom", outCustom);
			_parameters.Add("outServerName", outServerName);
			_parameters.Add("outServerPort", outServerPort);
			_parameters.Add("outUseSSL", outUseSSL);
			_parameters.Add("outUserName", outUserName);
			_parameters.Add("outPassword", outPassword);
			_parameters.Add("mailingListActive", mailingListActive);
			_parameters.Add("allowAnonymousEmail", allowAnonymousEmail);
			_parameters.Add("serviceContext", serviceContext);

			var _command = new JsonObject()
			{
				{ "/mbcategory/add-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> AddCategoryAsync(long userId, long parentCategoryId, string name, string description, IDictionary<string, object> serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("parentCategoryId", parentCategoryId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("serviceContext", serviceContext);

			var _command = new JsonObject()
			{
				{ "/mbcategory/add-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task DeleteCategoryAsync(long categoryId, bool includeTrashedEntries)
		{
			var _parameters = new JsonObject();

			_parameters.Add("categoryId", categoryId);
			_parameters.Add("includeTrashedEntries", includeTrashedEntries);

			var _command = new JsonObject()
			{
				{ "/mbcategory/delete-category", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task DeleteCategoryAsync(long groupId, long categoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("categoryId", categoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/delete-category", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<IDictionary<string, object>> GetCategoriesAsync(long groupId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> GetCategoriesAsync(long groupId, int status)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("status", status);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> GetCategoriesAsync(long groupId, long parentCategoryId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentCategoryId", parentCategoryId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> GetCategoriesAsync(long groupId, IEnumerable<long> parentCategoryIds, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentCategoryIds", parentCategoryIds);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> GetCategoriesAsync(long groupId, long parentCategoryId, int status, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentCategoryId", parentCategoryId);
			_parameters.Add("status", status);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> GetCategoriesAsync(long groupId, IEnumerable<long> parentCategoryIds, int status, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentCategoryIds", parentCategoryIds);
			_parameters.Add("status", status);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<long> GetCategoriesCountAsync(long groupId, long parentCategoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentCategoryId", parentCategoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCategoriesCountAsync(long groupId, IEnumerable<long> parentCategoryIds)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentCategoryIds", parentCategoryIds);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCategoriesCountAsync(long groupId, long parentCategoryId, int status)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentCategoryId", parentCategoryId);
			_parameters.Add("status", status);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<long> GetCategoriesCountAsync(long groupId, IEnumerable<long> parentCategoryIds, int status)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("parentCategoryIds", parentCategoryIds);
			_parameters.Add("status", status);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-categories-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IDictionary<string, object>> GetCategoryAsync(long categoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("categoryId", categoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IEnumerable<long>> GetCategoryIdsAsync(long groupId, long categoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("categoryId", categoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-category-ids", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			var _jsonArray = (JsonArray)_obj;

			return _jsonArray.Cast<long>();
		}

		public async Task<IDictionary<string, object>> GetSubcategoryIdsAsync(IEnumerable<object> categoryIds, long groupId, long categoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("categoryIds", categoryIds);
			_parameters.Add("groupId", groupId);
			_parameters.Add("categoryId", categoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-subcategory-ids", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> GetSubscribedCategoriesAsync(long groupId, long userId, int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("userId", userId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-subscribed-categories", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<long> GetSubscribedCategoriesCountAsync(long groupId, long userId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("userId", userId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/get-subscribed-categories-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IDictionary<string, object>> MoveCategoryAsync(long categoryId, long parentCategoryId, bool mergeWithParentCategory)
		{
			var _parameters = new JsonObject();

			_parameters.Add("categoryId", categoryId);
			_parameters.Add("parentCategoryId", parentCategoryId);
			_parameters.Add("mergeWithParentCategory", mergeWithParentCategory);

			var _command = new JsonObject()
			{
				{ "/mbcategory/move-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> MoveCategoryFromTrashAsync(long categoryId, long newCategoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("categoryId", categoryId);
			_parameters.Add("newCategoryId", newCategoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/move-category-from-trash", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task<IDictionary<string, object>> MoveCategoryToTrashAsync(long categoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("categoryId", categoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/move-category-to-trash", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}

		public async Task RestoreCategoryFromTrashAsync(long categoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("categoryId", categoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/restore-category-from-trash", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task SubscribeCategoryAsync(long groupId, long categoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("categoryId", categoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/subscribe-category", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task UnsubscribeCategoryAsync(long groupId, long categoryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("categoryId", categoryId);

			var _command = new JsonObject()
			{
				{ "/mbcategory/unsubscribe-category", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<IDictionary<string, object>> UpdateCategoryAsync(long categoryId, long parentCategoryId, string name, string description, string displayStyle, string emailAddress, string inProtocol, string inServerName, int inServerPort, bool inUseSSL, string inUserName, string inPassword, int inReadInterval, string outEmailAddress, bool outCustom, string outServerName, int outServerPort, bool outUseSSL, string outUserName, string outPassword, bool mailingListActive, bool allowAnonymousEmail, bool mergeWithParentCategory, IDictionary<string, object> serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("categoryId", categoryId);
			_parameters.Add("parentCategoryId", parentCategoryId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("displayStyle", displayStyle);
			_parameters.Add("emailAddress", emailAddress);
			_parameters.Add("inProtocol", inProtocol);
			_parameters.Add("inServerName", inServerName);
			_parameters.Add("inServerPort", inServerPort);
			_parameters.Add("inUseSSL", inUseSSL);
			_parameters.Add("inUserName", inUserName);
			_parameters.Add("inPassword", inPassword);
			_parameters.Add("inReadInterval", inReadInterval);
			_parameters.Add("outEmailAddress", outEmailAddress);
			_parameters.Add("outCustom", outCustom);
			_parameters.Add("outServerName", outServerName);
			_parameters.Add("outServerPort", outServerPort);
			_parameters.Add("outUseSSL", outUseSSL);
			_parameters.Add("outUserName", outUserName);
			_parameters.Add("outPassword", outPassword);
			_parameters.Add("mailingListActive", mailingListActive);
			_parameters.Add("allowAnonymousEmail", allowAnonymousEmail);
			_parameters.Add("mergeWithParentCategory", mergeWithParentCategory);
			_parameters.Add("serviceContext", serviceContext);

			var _command = new JsonObject()
			{
				{ "/mbcategory/update-category", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IDictionary<string, object>)_obj;
		}
	}
}